source/common/api/win32/os_sys_calls_impl.cc

```diff
@@ -244,12 +244,20 @@ SysCallIntResult OsSysCallsImpl::fstat(os_fd_t fd, struct stat* buf) {
 
 SysCallIntResult OsSysCallsImpl::setsockopt(os_fd_t sockfd, int level, int optname,
                                             const void* optval, socklen_t optlen) {
+  if (optname == IP_RECVTOS || optname == IPV6_RECVTCLASS) {
```

Comment:
Am I understanding correctly that we need to use `WSASetRecvIPEcn` instead of `setsockopt` only for IPv6? For IPv4, we still use `setsockopt`?

Reply:
This is a misunderstanding, discussed offline.